public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -0,0 +1,248 @@
+import React, { FC, useCallback, useEffect, useMemo, useReducer, useRef, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+
+import { LoadingState, PanelData } from '@grafana/data';
+import { selectors } from '@grafana/e2e-selectors';
+import { config } from '@grafana/runtime';
+import { Alert, Button, Field, InputControl, Stack, Tooltip } from '@grafana/ui';
+import { isExpressionQuery } from 'app/features/expressions/guards';
+import { AlertQuery } from 'app/types/unified-alerting-dto';
+
+import { AlertingQueryRunner } from '../../../state/AlertingQueryRunner';
+import { RuleFormType, RuleFormValues } from '../../../types/rule-form';
+import { getDefaultOrFirstCompatibleDataSource } from '../../../utils/datasource';
+import { ExpressionEditor } from '../ExpressionEditor';
+import { ExpressionsEditor } from '../ExpressionsEditor';
+import { QueryEditor } from '../QueryEditor';
+import { RuleEditorSection } from '../RuleEditorSection';
+import { refIdExists } from '../util';
+
+import { AlertType } from './AlertType';
+import {
+  addNewDataQuery,
+  addNewExpression,
+  queriesAndExpressionsReducer,
+  removeExpression,
+  rewireExpressions,
+  setDataQueries,
+  updateExpression,
+  updateExpressionRefId,
+  updateExpressionType,
+} from './reducer';
+
+interface Props {
+  editingExistingRule: boolean;
+}
+
+export const QueryAndExpressionsStep: FC<Props> = ({ editingExistingRule }) => {
```

Comment:
💬 This component has gotten quite large but I'm not sure if adding additional abstractions is a good idea at this time, we might enter the realm of diminishing returns by adding too many abstractions and passing everything as props or function args.

Open to feedback on this one for sure!